src/MessagingClient.ts

```diff
@@ -387,22 +387,26 @@ export class MessagingClient implements MessagingAPI {
         return this.getRoomInformation(room).conversation
     }
 
-    async searchChannel(searchTerm: string, limit: number, since?: string): Promise<SearchChannelsResponse> {
+    async searchChannel(limit: number, searchTerm?: string, since?: string): Promise<SearchChannelsResponse> {
         try {
             let publicRooms: Array<IPublicRoomsChunkRoom> = []
             let res: IPublicRoomsResponse
             let nextBatch = since
+            let options = searchTerm ? {
+                filter: {
+                  generic_search_term: searchTerm
+                },
+                limit,
+                since: nextBatch
+            } : {
+                limit,
+                since: nextBatch
+            }
```

Comment:
nit: How about sth like this?
```suggestion
            const filter = searchTerm ? { filter: {
                  generic_search_term: searchTerm
                }
              } : {}
            const options = {
                limit,
                since: nextBatch
                ...filter
            }
```